Sources/HarmonizeSemantics/GrammarComponents/Decoration/FunctionCall.swift

```diff
@@ -109,6 +116,10 @@ public extension FunctionCall {
         public let value: String
     }
 
+    /// A struct representing a single token that may be composing a function call.
+    ///
+    /// For example, in the function call `greet().$someObj.flatMap { }.`, we have two tokens: `$someObjc` and `.flatMap`
+    /// We still need to extend this in the future so that it is fully typed since a Token can be another FunctionCall or a reference and so on.
```

Comment:
I believe `greet` is also a token in AST's. Basically nodes are composed of smaller tokens.